consensus/spos/bls/subroundSignature_test.go

```diff
@@ -650,62 +862,74 @@ func TestSubroundSignature_DoSignatureConsensusCheckShouldReturnTrueWhenSignatur
 		_ = sr.SetJobDone(sr.ConsensusGroup()[i], bls.SrSignature, true)
 	}
 
+	sr.Header = &block.HeaderV2{}
 	assert.True(t, sr.DoSignatureConsensusCheck())
 }
 
 func TestSubroundSignature_DoSignatureConsensusCheckShouldReturnFalseWhenSignaturesCollectedReturnFalse(t *testing.T) {
 	t.Parallel()
 
 	sr := *initSubroundSignature()
+	sr.Header = &block.HeaderV2{Header: createDefaultHeader()}
 	assert.False(t, sr.DoSignatureConsensusCheck())
 }
 
-func TestSubroundSignature_DoSignatureConsensusCheckShouldReturnFalseWhenNotAllSignaturesCollectedAndTimeIsNotOut(t *testing.T) {
+func TestSubroundSignature_DoSignatureConsensusCheckNotAllSignaturesCollectedAndTimeIsNotOut(t *testing.T) {
 	t.Parallel()
 
-	container := mock.InitConsensusCore()
-	sr := *initSubroundSignatureWithContainer(container)
-	sr.WaitingAllSignaturesTimeOut = false
-
-	sr.SetSelfPubKey(sr.ConsensusGroup()[0])
-
-	for i := 0; i < sr.Threshold(bls.SrSignature); i++ {
-		_ = sr.SetJobDone(sr.ConsensusGroup()[i], bls.SrSignature, true)
-	}
-
-	assert.False(t, sr.DoSignatureConsensusCheck())
+	t.Run("with flag active, should return false - will be done on subroundEndRound", testSubroundSignatureDoSignatureConsensusCheck(true, setThresholdJobsDone, false, false))
```

Comment:
can you have an arg struct for the argument, then it will be clearer which fields are false and which are true.

Reply:
pushed, thanks